### PR TITLE
add time services support

### PR DESCRIPTION
**Pull request by trevharmon**
This commit uses the `W32tm` service to sync the clock of the system.
It does require the current time be within a certain range, otherwise
it will complain and ask for manual intervention.

The service is stopped, the new setting are applied, and then a time
sync is forced once the service has been restarted. If this fails,
the script waits several seconds and then retries the action. If it
fails a second time, the script aborts with an error message.

### CODE REVIEW

**Comment by trevharmon** [comment]
Contains all changes of PR #14

**Comment by natemccurdy** [comment]
Please rebase off of `master` now that the cleanup PR has been merged.

**Comment by natemccurdy** [comment]
@trevharmon To keep the footprint of this script as low as possible by default, can you restructure the code such that configuring NTP is **disabled** by default?

File: README.md [hunk 1]
```diff
@@ -166,6 +167,11 @@ An array of FQDN NTP servers to use to do the time sync.
 
 Default value: `["0.pool.ntp.org","1.pool.ntp.org","2.pool.ntp.org"]`
 
+#### `set_ntp_serves`
+Whether or not to the NTP servers.
```
Comment (natemccurdy) [inline README.md]: missing a word?

**Comment by trevharmon** [comment]
Configuring NTP is now disabled by default via the `set_ntp_servers` parameter.

**Comment by natemccurdy** [comment]
@trevharmon So, after thinking about your PR's, I've come to the conclusion that the logic to set DNS or NTP should be handled by flags to the powershell script; they should not be configured by parameters to the Puppet code.

By making these parameters to Puppet code, the powershell script can only ever be used in one way. This is not ideal in the case where you want to install the Puppet agent on many Windows boxes where some might need DNS, some might need NTP, some might need both, and some might need none.

If you think you can rewrite the Powershell to handle all the logic, then please submit a new PR to do so.
If not, I think this is a good feature, and I will try to implement it myself.

Let me know what you think. I'm planning on closing this PR and #16 as well, but I'll keep them open for a bit to have a place to converse.

**Comment by trevharmon** [comment]
Will resubmit PR once it's cleaned up.

**Comment by natemccurdy** [comment]
Sounds good @trevharmon 👍 
Thanks for working on this.

**Comment by trevharmon** [comment]
See #17

**Pull request closed without merging**